### PR TITLE
🐞 Destination S3 & GCS Avro: support array with unknown item type

### LINKED ISSUE #9252: 🐛 Source Slack: more specific typing required for `pending_shared` field
Is this your first time deploying Airbyte: Yes
OS Version / Instance: EC2 t2.large
Memory / Disk: 8Gb / 40 gb
Deployment: Docker
Airbyte Version: 0.34.4-alpha
Source name/version: source-slack:0.1.13
Destination name/version: destination-s3:0.2.0
Step: On sync
Description: Conversion of `channel_messages` to Parquet, Avro or JSON fails because the items of the field `pending_shared` have no type:

````
2021-12-29 05:56:47 [43mdestination[0m > Exception in thread "main" java.lang.IllegalStateException: Field pending_shared.items has no type
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.getTypes(JsonToAvroSchemaConverter.java:63)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.getNonNullTypes(JsonToAvroSchemaConverter.java:51)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.getNullableFieldTypes(JsonToAvroSchemaConverter.java:237)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.getSingleFieldType(JsonToAvroSchemaConverter.java:200)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.lambda$getNullableFieldTypes$4(JsonToAvroSchemaConverter.java:240)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:273)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2021-12-29 05:56:47 [43mdestination[0m > 	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.getNullableFieldTypes(JsonToAvroSchemaConverter.java:248)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.avro.JsonToAvroSchemaConverter.getAvroSchema(JsonToAvroSchemaConverter.java:153)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.writer.ProductionWriterFactory.create(ProductionWriterFactory.java:40)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.S3Consumer.startTracked(S3Consumer.java:53)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.base.FailureTrackingAirbyteMessageConsumer.start(FailureTrackingAirbyteMessageConsumer.java:34)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.base.IntegrationRunner.consumeWriteStream(IntegrationRunner.java:142)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:128)
2021-12-29 05:56:47 [43mdestination[0m > 	at io.airbyte.integrations.destination.s3.S3Destination.main(S3Destination.java:30)
````

The sync works with CSV (though there is a warning on schema validation). Parquet, Avro, JSON are failing.

[Related slack conversation
]( https://airbytehq.slack.com/archives/C01MFR03D5W/p1640758132458900?thread_ts=1640758132.458900&cid=C01MFR03D5W)

### Logs:
[logs-507-0.txt](https://github.com/airbytehq/airbyte/files/7803031/logs-507-0.txt)

### PR DESCRIPTION
**Pull request by tuliren**
## What
- When the array field has an `items` property, but the property is empty (i.e. without a `type` specification), the Json to Avro schema converter will throw exception.
- This PR defaults such array to an array of strings.
- **This PR also defaults any kind of exception to the string schema. We should not see errors from the Json to Avro schema converter any more.**
- Resolve #9252.

## 🚨 User Impact 🚨
This PR should not affect any existing connections.

## Pre-merge Checklist

<strong>Updating a connector</strong>
   
#### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [x] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

### CODE REVIEW

**Comment by tuliren** [comment]
Integration test failed due to `Could not find image: airbyte/normalization:dev`. I wonder if this is due to insufficient disk space of our Github runner.

**Review by edgao — APPROVED** [review]
Should the json/avro conversion doc also mention that any errors will default to string-type? (everything else seems fine!)

**Review by alafanechere — APPROVED** [review]
LGTM, thank you for the fix!

File: airbyte-config/init/src/main/resources/seed/destination_specs.yaml [hunk 1]
```diff
@@ -3594,8 +3594,6 @@
                   \ more memory. Allowed values: min=5MB, max=525MB Default: 5MB."
                 type: "integer"
                 default: 5
-                minimum: 5
-                maximum: 525
```
Comment (tuliren) [inline airbyte-config/init/src/main/resources/seed/destination_specs.yaml:3598]: I am really confused why the spec changed. I did not touch the `spec.json` at all.

**Pull request merged**